src/dlload.c

```diff
@@ -139,26 +136,20 @@ static void *jl_load_dynamic_library_(const char *modname, unsigned flags, int t
 #endif
         goto done;
     }
-    /*
-      this branch shortcuts absolute paths
-    */
+
 #ifdef _OS_WINDOWS_
-    else if (modname[1] == ':') {
```

Comment:
By the way, I noticed that this line was both incorrect and a potential buffer overrun if `modname` is an empty string.  My new version checks that `modname[0]` is nonzero.